### ISSUE TITLE
Problems with cg deliver balsamic fastq files

### ISSUE THREAD

**Issue by ashwini06** [Clinical-Genomics/cg#1259]
[](url)## Description

According to the balsamic amdoc 2046, both multiqc and fastq, should be delivered with the `balsamic-qc` flag `. But noticed problems with the delivery of fastq files.

`cg deliver analysis -c firstthrush -d balsamic-qc` does not deliver fastq files to the customer folder `/home/proj/production/customers/cust125/inbox/783655/T-1/`

## This can be closed when

When both multiqc and fastqc are delivered with `balsamic-qc` flag by `cg deliver ....` command

<img width="322" alt="aa" src="https://user-images.githubusercontent.com/6302819/131331369-5a062b5a-d2b2-41a2-b4c7-13f82abe77ba.png">

**Comment by henrikstranneheim** [comment]
@ashwini06 Could you add some more detail to this issue?

**Comment by henrikstranneheim** [comment]
I do not think there is an issue with the code:
Both fastq and multiqc tags is defined in cg/constants/delivery.py:
```
BALSAMIC_QC_CASE_TAGS = [
    {"multiqc-html"},
]
BALSAMIC_QC_SAMPLE_TAGS = [
    {"fastq"},
]
```
For case `firstthrush` there is no `fastq` tag but a similiar `quality-trimmed-fastq-read1 and quality-trimmed-fastq-read2` tag

```
[0|0|0] [henrik.stranneheim@hasta:~] [P_main] 6s $ housekeeper get bundle -v firstthrush
2022-03-09 11:37:21 hasta.scilifelab.se housekeeper.cli.core[178458] INFO Use database mysql+pymysql://produser:ZkfQm9.Ww@127.0.0.1:19002/housekeeper
2022-03-09 11:37:21 hasta.scilifelab.se housekeeper.cli.core[178458] INFO Use root path /home/proj/production/housekeeper-bundles
2022-03-09 11:37:21 hasta.scilifelab.se housekeeper.store.api.find[178458] INFO Fetching all bundles
2022-03-09 11:37:22 hasta.scilifelab.se housekeeper.store.api.find[178458] INFO Fetching bundle with name: firstthrush
                📦 Bundle table 📦
┏━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━━━┓
┃ ID    ┃ Bundle name ┃ Version IDs ┃ Created    ┃
┡━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━━━━━━┩
│ 27426 │ firstthrush │ 29085       │ 2020-12-21 │
└───────┴─────────────┴─────────────┴────────────┘
2022-03-09 11:37:22 hasta.scilifelab.se housekeeper.store.api.find[178458] INFO Fetching bundle with id: 27426
                               📕 Version table 📕
┏━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━━━━┳━━━━━━━━━┓
┃ ID    ┃ Bundle name ┃ Nr files ┃ Included   ┃ Archived ┃ Created    ┃ Expires ┃
┡━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━━━━╇━━━━━━━━━┩
│ 29085 │ firstthrush │ 57       │ 2020-12-22 │          │ 2020-12-21 │         │
└───────┴─────────────┴──────────┴────────────┴──────────┴────────────┴─────────┘
2022-03-09 11:37:22 hasta.scilifelab.se housekeeper.store.api.find[178458] INFO Fetching files from version 29085
                                                                       📜 Files table 📜
┏━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ ID     ┃ File name                                                               ┃ Tags                                                                     ┃
┡━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│ 631378 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SV.so… │ manta, sv, vcf-all, annotated-somatic-vcf-all, firstthrush               │
│ 631379 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SV.so… │ manta, sv, vcf-all, annotated-somatic-vcf-all-index, firstthrush         │
│ 631380 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/CNV.s… │ cnv, cnvkit, vcf-all, annotated-somatic-vcf-all, firstthrush             │
│ 631381 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/CNV.s… │ cnv, cnvkit, vcf-all, annotated-somatic-vcf-all-index, firstthrush       │
│ 631382 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.s… │ vardict, snv, vcf-all, annotated-somatic-vcf-all, firstthrush            │
│ 631383 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.s… │ vardict, snv, vcf-all, annotated-somatic-vcf-all-index, firstthrush      │
│ 631384 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.s… │ snv, mutect, vcf-all, annotated-somatic-vcf-all, firstthrush             │
│ 631385 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.s… │ snv, mutect, vcf-all, annotated-somatic-vcf-all-index, firstthrush       │
│ 631386 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.s… │ snv, tnhaplotyper, vcf-all, annotated-somatic-vcf-all, firstthrush       │
│ 631387 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.s… │ snv, tnhaplotyper, vcf-all, annotated-somatic-vcf-all-index, firstthrush │
│ 631388 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.s… │ vardict, snv, vcf-summary, annotated-somatic-vcf-summary, firstthrush    │
│ 631389 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.s… │ snv, mutect, vcf-summary, annotated-somatic-vcf-summary, firstthrush     │
│ 631390 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.s… │ snv, tnhaplotyper, vcf-summary, annotated-somatic-vcf-summary,           │
│        │                                                                         │ firstthrush                                                              │
│ 631391 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/CNV.s… │ cnv, cnvkit, vcf-summary, annotated-somatic-vcf-summary, firstthrush     │
│ 631392 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SV.so… │ manta, sv, vcf-summary, annotated-somatic-vcf-summary, firstthrush       │
│ 631393 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SV.so… │ manta, sv, annotated-somatic-vcf-pass, vcf-pass, firstthrush             │
│ 631394 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SV.so… │ manta, sv, annotated-somatic-vcf-pass-index, vcf-pass, firstthrush       │
│ 631395 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.s… │ vardict, snv, annotated-somatic-vcf-pass, vcf-pass, firstthrush          │
│ 631396 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.s… │ vardict, snv, annotated-somatic-vcf-pass-index, vcf-pass, firstthrush    │
│ 631397 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.s… │ snv, tnhaplotyper, annotated-somatic-vcf-pass, vcf-pass, firstthrush     │
│ 631398 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.s… │ snv, tnhaplotyper, annotated-somatic-vcf-pass-index, vcf-pass,           │
│        │                                                                         │ firstthrush                                                              │
│ 631399 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/CNV.s… │ cnv, annotated-somatic-vcf-pass, cnvkit, vcf-pass, firstthrush           │
│ 631400 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/CNV.s… │ cnv, annotated-somatic-vcf-pass-index, cnvkit, vcf-pass, firstthrush     │
│ 631401 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.s… │ snv, mutect, annotated-somatic-vcf-pass, vcf-pass, firstthrush           │
│ 631402 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.s… │ snv, mutect, annotated-somatic-vcf-pass-index, vcf-pass, firstthrush     │
│ 631403 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/multi… │ multiqc-html, html                                                       │
│ 631404 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/multi… │ multiqc-json, json                                                       │
│ 631405 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/tumor… │ cnv-cns, cns                                                             │
│ 631406 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/tumor… │ cnv-cnr, cnr                                                             │
│ 631407 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/tumor… │ cnv-scatter, scatter                                                     │
│ 631408 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/tumor… │ cnv-diagram, diagram                                                     │
│ 631409 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/first… │ gene-breaks, cnv-gene-breaks                                             │
│ 631410 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/first… │ gene-metrics, cnv-gene-metrics                                           │
│ 631411 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.s… │ snv, vcf-filtered, clinical-vcf-filtered, firstthrush                    │
│ 631412 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.s… │ snv, vcf-filtered, clinical-vcf-filtered-index, firstthrush              │
│ 631413 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.s… │ snv, vcf-pass, clinical-vcf-pass, firstthrush                            │
│ 631414 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.s… │ snv, vcf-pass, clinical-vcf-pass-index, firstthrush                      │
│ 631415 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.s… │ vardict, tmb, stat-somatic-tmb, firstthrush                              │
│ 631420 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.g… │ snv, vcf-all, annotated-germline-vcf-all, haplotypecaller, ACC6365A1     │
│ 631421 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.g… │ snv, vcf-all, annotated-germline-vcf-all-index, haplotypecaller,         │
│        │                                                                         │ ACC6365A1                                                                │
│ 631422 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SV.ge… │ sv, manta-germline, vcf-all, annotated-germline-vcf-all, ACC6365A1       │
│ 631423 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SV.ge… │ sv, manta-germline, vcf-all, annotated-germline-vcf-all-index, ACC6365A1 │
│ 631424 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.g… │ snv, vcf-all, annotated-germline-vcf-all, strelka-germline, ACC6365A1    │
│ 631425 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.g… │ snv, vcf-all, annotated-germline-vcf-all-index, strelka-germline,        │
│        │                                                                         │ ACC6365A1                                                                │
│ 631426 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.g… │ snv, vcf-all, annotated-germline-vcf-all, dnascope, ACC6365A1            │
│ 631427 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.g… │ snv, vcf-all, annotated-germline-vcf-all-index, dnascope, ACC6365A1      │
│ 631428 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SV.ge… │ sv, manta-germline, vcf-summary, annotated-germline-vcf-summary,         │
│        │                                                                         │ ACC6365A1                                                                │
│ 631429 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.g… │ snv, dnascope, vcf-summary, annotated-germline-vcf-summary, ACC6365A1    │
│ 631430 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.g… │ snv, haplotypecaller, vcf-summary, annotated-germline-vcf-summary,       │
│        │                                                                         │ ACC6365A1                                                                │
│ 631431 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/SNV.g… │ snv, strelka-germline, vcf-summary, annotated-germline-vcf-summary,      │
│        │                                                                         │ ACC6365A1                                                                │
│ 631432 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/conca… │ read1, quality-trimmed-fastq-read1, ACC6365A1                            │
│ 631433 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/conca… │ read2, quality-trimmed-fastq-read2, ACC6365A1                            │
│ 631434 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/conca… │ json, quality-trimmed-fastq-json, ACC6365A1                              │
│ 631435 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/conca… │ html, quality-trimmed-fastq-html, ACC6365A1                              │
│ 631436 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/first… │ balsamic-report                                                          │
│ 631437 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/first… │ balsamic-config                                                          │
│ 631438 │ /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/first… │ balsamic-dag                                                             │
└────────┴─────────────────────────────────────────────────────────────────────────┴──────────────────────────────────────────────────────────────────────────┘

[0|0|0] [henrik.stranneheim@hasta:~] [P_main] 2s $ cg deliver analysis -c firstthrush -d balsamic-qc --dry-run
Running CG deliver
Set dry run to True
Fetch latest version from bundle firstthrush
Setting ticket_id to 783655
Setting customer_id to cust125
Would hard link file /home/proj/production/housekeeper-bundles/firstthrush/2020-12-21/multiqc_report.html to /home/proj/production/customers/cust125/inbox/783655/T-1/multiqc_report.html
Linked 1 files for case firstthrush
Linked 0 files for sample ACC6365A1, case firstthrush
```

For a random recent case: `dearmayfly` there is `fastq` tags and they are also delivered:

```

[0|0|0] [henrik.stranneheim@hasta:~] [P_main] 6s $ housekeeper get bundle -v dearmayfly
2022-03-09 11:36:00 hasta.scilifelab.se housekeeper.cli.core[176515] INFO Use database mysql+pymysql://produser:ZkfQm9.Ww@127.0.0.1:19002/housekeeper
2022-03-09 11:36:00 hasta.scilifelab.se housekeeper.cli.core[176515] INFO Use root path /home/proj/production/housekeeper-bundles
2022-03-09 11:36:00 hasta.scilifelab.se housekeeper.store.api.find[176515] INFO Fetching all bundles
2022-03-09 11:36:00 hasta.scilifelab.se housekeeper.store.api.find[176515] INFO Fetching bundle with name: dearmayfly
                📦 Bundle table 📦
┏━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━━━┓
┃ ID    ┃ Bundle name ┃ Version IDs ┃ Created    ┃
┡━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━━━━━━┩
│ 83372 │ dearmayfly  │ 85858       │ 2022-03-07 │
└───────┴─────────────┴─────────────┴────────────┘
2022-03-09 11:36:01 hasta.scilifelab.se housekeeper.store.api.find[176515] INFO Fetching bundle with id: 83372
                               📕 Version table 📕
┏━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━━━━┳━━━━━━━━━━┳━━━━━━━━━━━━┳━━━━━━━━━┓
┃ ID    ┃ Bundle name ┃ Nr files ┃ Included   ┃ Archived ┃ Created    ┃ Expires ┃
┡━━━━━━━╇━━━━━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━━━━╇━━━━━━━━━━╇━━━━━━━━━━━━╇━━━━━━━━━┩
│ 85858 │ dearmayfly  │ 64       │ 2022-03-07 │          │ 2022-03-07 │         │
└───────┴─────────────┴──────────┴────────────┴──────────┴────────────┴─────────┘
2022-03-09 11:36:01 hasta.scilifelab.se housekeeper.store.api.find[176515] INFO Fetching files from version 85858
                                                                       📜 Files table 📜
┏━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ ID      ┃ File name                                                               ┃ Tags                                                                    ┃
┡━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│ 1865010 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.ge… │ vcf, storage, haplotype-caller, germline, ACC9493A12                    │
│ 1865011 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.ge… │ storage, vcf-index, haplotype-caller, germline, ACC9493A12              │
│ 1865012 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SV.ger… │ manta, storage, sv-vcf, germline, ACC9493A12                            │
│ 1865013 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SV.ger… │ manta, storage, sv-vcf-index, germline, ACC9493A12                      │
│ 1865014 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.ge… │ vcf, dnascope, storage, germline, ACC9493A12                            │
│ 1865015 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.ge… │ dnascope, storage, vcf-index, germline, ACC9493A12                      │
│ 1865016 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.ge… │ haplotype-caller, vcf-report, germline, ACC9493A12                      │
│ 1865017 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.ge… │ dnascope, vcf-report, germline, ACC9493A12                              │
│ 1865018 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SV.ger… │ manta, vcf-report, germline, ACC9493A12                                 │
│ 1865019 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SV.som… │ vcf-sv-clinical, manta, scout, deliver, filtered, dearmayfly            │
│ 1865020 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SV.som… │ vcf-sv-clinical-index, manta, scout, deliver, filtered, dearmayfly      │
│ 1865021 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/CNV.so… │ vcf-sv-research, cnvkit, deliver, filtered, dearmayfly                  │
│ 1865022 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/CNV.so… │ vcf-sv-research-index, cnvkit, deliver, filtered, dearmayfly            │
│ 1865023 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/tumor.… │ cnvkit, deliver, segments, dearmayfly                                   │
│ 1865024 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/tumor.… │ cnvkit, regions, deliver, dearmayfly                                    │
│ 1865025 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/tumor.… │ cnvkit, diagram, deliver, visualization, dearmayfly                     │
│ 1865026 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/dearma… │ cnvkit, genes, storage, dearmayfly                                      │
│ 1865027 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/dearma… │ cnvkit, metrics, deliver, genes, dearmayfly                             │
│ 1865028 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/tumor.… │ cnvkit, deliver, visualization, dearmayfly                              │
│ 1865029 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.so… │ vcf, vardict, storage, somatic, dearmayfly                              │
│ 1865030 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.so… │ vardict, storage, vcf-index, somatic, dearmayfly                        │
│ 1865031 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.so… │ vcf, deliver, storage, somatic, tnscope-umi, dearmayfly                 │
│ 1865032 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.so… │ deliver, storage, vcf-index, somatic, tnscope-umi, dearmayfly           │
│ 1865033 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.so… │ vcf, storage, tumor, haplotype-caller, somatic, dearmayfly              │
│ 1865034 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.so… │ storage, tumor, vcf-index, haplotype-caller, somatic, dearmayfly        │
│ 1865035 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SV.som… │ vcf, deliver, somatic, delly, dearmayfly                                │
│ 1865036 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SV.som… │ deliver, vcf-index, somatic, delly, dearmayfly                          │
│ 1865037 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/CNV.so… │ cnvkit, deliver, tumor, sv-vcf, somatic, dearmayfly                     │
│ 1865038 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/CNV.so… │ cnvkit, deliver, tumor, sv-vcf-index, somatic, dearmayfly               │
│ 1865039 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SV.som… │ vcf-sv-research, manta, scout, deliver, tumor, somatic, dearmayfly      │
│ 1865040 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SV.som… │ vcf-sv-research-index, manta, scout, deliver, tumor, somatic,           │
│         │                                                                         │ dearmayfly                                                              │
│ 1865041 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.so… │ vardict, vcf-report, somatic, dearmayfly                                │
│ 1865042 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.so… │ haplotype-caller, sention, vcf-report, somatic, dearmayfly              │
│ 1865043 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SV.som… │ manta, sention, vcf-report, somatic, dearmayfly                         │
│ 1865044 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.so… │ vcf-report, somatic, tnscope-umi, dearmayfly                            │
│ 1865045 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/CNV.so… │ cnvkit, vcf-report, somatic, dearmayfly                                 │
│ 1865046 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.so… │ vcf-snv-research, storage, haplotype-caller, filtered, somatic,         │
│         │                                                                         │ dearmayfly                                                              │
│ 1865047 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.so… │ vcf-snv-research-index, storage, haplotype-caller, filtered, somatic,   │
│         │                                                                         │ dearmayfly                                                              │
│ 1865048 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.so… │ vcf, storage, haplotype-caller, filtered, somatic, dearmayfly           │
│ 1865049 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.so… │ storage, vcf-index, haplotype-caller, filtered, somatic, dearmayfly     │
│ 1865050 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/tumor.… │ cram, scout, deliver, tumor, ACC9493A12                                 │
│ 1865051 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/tumor.… │ cram-index, scout, deliver, tumor, ACC9493A12                           │
│ 1865052 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SV.som… │ vcf-sv-research, deliver, filtered, dearmayfly                          │
│ 1865053 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SV.som… │ vcf-sv-research-index, deliver, filtered, dearmayfly                    │
│ 1865054 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.so… │ vcf, storage, vcf-snv-filtered, somatic, dearmayfly                     │
│ 1865055 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.so… │ storage, vcf-index, vcf-snv-filtered-index, somatic, dearmayfly         │
│ 1865056 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.so… │ vcf-snv-clinical, vcf, scout, deliver, somatic, dearmayfly              │
│ 1865057 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.so… │ vcf-snv-clinical-index, scout, deliver, vcf-index, somatic, dearmayfly  │
│ 1865058 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/multiq… │ multiqc-html, scout, deliver, dearmayfly                                │
│ 1865059 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/multiq… │ multiqc-json, vogue, dearmayfly                                         │
│ 1865060 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.so… │ vcf-snv-research, storage, haplotype-caller, filtered, somatic,         │
│         │                                                                         │ dearmayfly                                                              │
│ 1865061 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.so… │ vcf-snv-research-index, storage, haplotype-caller, filtered, somatic,   │
│         │                                                                         │ dearmayfly                                                              │
│ 1865062 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.so… │ vcf, storage, haplotype-caller, filtered, somatic, dearmayfly           │
│ 1865063 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/SNV.so… │ storage, vcf-index, haplotype-caller, filtered, somatic, dearmayfly     │
│ 1865064 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/concat… │ fastq, deliver, ACC9493A12                                              │
│ 1865065 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/concat… │ fastq, deliver, ACC9493A12                                              │
│ 1865066 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/concat… │ fastq, metrics, ACC9493A12                                              │
│ 1865067 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/concat… │ fastq, visualization, ACC9493A12                                        │
│ 1865068 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/dearma… │ balsamic-report, dearmayfly                                             │
│ 1865069 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/dearma… │ balsamic-config, dearmayfly                                             │
│ 1865070 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/dearma… │ balsamic-dag, dearmayfly                                                │
│ 1865071 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/dearma… │ delivery-report, scout, dearmayfly                                      │
│ 1865072 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/dearma… │ deliver, vogue, qc-metrics, deliverable, dearmayfly                     │
│ 1866193 │ /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/scout_… │ scout-load-config                                                       │
└─────────┴─────────────────────────────────────────────────────────────────────────┴─────────────────────────────────────────────────────────────────────────┘

[0|0|0] [henrik.stranneheim@hasta:~] [P_main] 20s $ cg deliver analysis -c dearmayfly -d balsamic-qc --dry-run
Running CG deliver
Set dry run to True
Fetch latest version from bundle dearmayfly
Setting ticket_id to 500382
Setting customer_id to cust087
Would hard link file /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/multiqc_report.html to /home/proj/production/customers/cust087/inbox/500382/F0043979-2/multiqc_report.html
Linked 1 files for case dearmayfly
Would hard link file /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/concatenated_ACC9493A12_XXXXXX_R_1.fp.fastq.gz to /home/proj/production/customers/cust087/inbox/500382/F0043979-2/2022-05451-04/concatenated_2022-05451-04_XXXXXX_R_1.fp.fastq.gz
Would hard link file /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/concatenated_ACC9493A12_XXXXXX_R_2.fp.fastq.gz to /home/proj/production/customers/cust087/inbox/500382/F0043979-2/2022-05451-04/concatenated_2022-05451-04_XXXXXX_R_2.fp.fastq.gz
Would hard link file /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/concatenated_ACC9493A12_XXXXXX_R_fastp.json to /home/proj/production/customers/cust087/inbox/500382/F0043979-2/2022-05451-04/concatenated_2022-05451-04_XXXXXX_R_fastp.json
Would hard link file /home/proj/production/housekeeper-bundles/dearmayfly/2022-03-07/concatenated_ACC9493A12_XXXXXX_R_fastp.html to /home/proj/production/customers/cust087/inbox/500382/F0043979-2/2022-05451-04/concatenated_2022-05451-04_XXXXXX_R_fastp.html
Linked 4 files for sample ACC9493A12, case dearmayfly
```

**Comment by henrikstranneheim** [comment]
@ashwini06 @patrikgrenfeldt I think these balsamic generated fastq files should be treated as temporary and be removed in the automation as they take up a lot of space.

@patrikgrenfeldt We should include the tag `quality-trimmed-fastq-read1` and `quality-trimmed-fastq-read2` in the clean-up of concatenated trimmed Balsamic files (if it is not already so)

**Comment by ashwini06** [comment]
> @ashwini06 @patrikgrenfeldt I think these balsamic generated fastq files should be treated as temporary and be removed in the automation as they take up a lot of space.

@henrikstranneheim:  are these adapter trimmed fastq files`concatenated_2022-05451-04_XXXXXX_R_*.fp.fastq.gz` not being stored/delivered to the customer? We have a discussion going on in this [issue](https://github.com/Clinical-Genomics/hermes/issues/44). If these files are not required, snakemake has the option to mark them as temp files and remove them automatically once the analysis is finished.

**Comment by henrikstranneheim** [comment]
@moahaegglund Correct me if I am wrong, but I think they are delivered but only stay in Housekeeper for 21 days and are then removed. However, the removal (I think) uses the `fastq` tag and files lacking `fastq` tag will currently not be removed. Hence, these files having only `quality-trimmed-fastq-read1` will not be removed in this process

**Comment by henrikstranneheim** [comment]
## Update:

After discussion with @keyvanelhami, @ashwini06, @ivadym  @moahaegglund We have decided to let Balsamic assign the trimmed concatenated files as temporary in snakemake. Hence they will be removed by Balsamic once the analysis is completed. We will only deliver "raw" fastq files to the customers and provide instructions on how to replicate the trimming steps.

@Mropat @patrikgrenfeldt This also means that we should remove the cg-clean-trimmed-fastqs.service (config/hasta.scilifelab.se/systemd/clean-trimmed-fastqs.service) once the changes in Balsamic are in place. @ashwini06 Is this part of the next Balsamic major release?

**Comment by ashwini06** [comment]
@ashwini06 Is this part of the next Balsamic major release?

@henrikstranneheim Yes, I added the[ balsamic issue](https://github.com/Clinical-Genomics/BALSAMIC/issues/911) to this fix for upcoming release

**Comment by henrikstranneheim** [comment]
@ashwini06 Great! I will close this one as no code needs to change in cg right now! Let's open a new issue to remove the systemd service file in servers once Balsamic has been updated in production.
